tools/sleigh-lift/src/main.cpp

```diff
@@ -12,10 +12,35 @@
 #include <iostream>
 #include <string>
 
-static void PrintUsage(void) {
-  std::cerr << "Usage: sleigh-lift [action] [sla_file] [bytes] [-a address] "
-               "[-p root_sla_dir] [-s pspec_file]"
-            << std::endl;
+static void PrintUsage(std::ostream &os) {
+  os << "Usage: sleigh-lift [action] [sla_file] [bytes] [-a address] "
+        "[-p root_sla_dir] [-s pspec_file]"
+     << std::endl;
+}
+
+static void PrintVersion(void) {
+  std::cout << "sleigh-lift " << sleigh::GetGhidraVersion() << '\n';
+
+  // Print out the commit info for the underlying GHIDRA checkout
+  std::cout << "GHIDRA Version: " << sleigh::GetGhidraVersion() << '\n'
+            << "GHIDRA Commit Hash: " << sleigh::GetGhidraCommitHash() << '\n';
+
+  // Now print out the Git commit information
+  if (sleigh::HasVersionData()) {
+    std::cout << "Commit Hash: " << sleigh::GetCommitHash() << '\n'
+              << "Commit Date: " << sleigh::GetCommitDate() << '\n'
+              << "Last commit by: " << sleigh::GetAuthorName() << " ["
+              << sleigh::GetAuthorEmail() << "]\n"
+              << "Commit Subject: [" << sleigh::GetCommitSubject() << "]\n"
+              << '\n';
+    if (sleigh::HasUncommittedChanges()) {
+      std::cout << "Uncommitted changes were present during build.\n";
+    } else {
+      std::cout << "All changes were committed prior to building.\n";
+    }
+  } else {
+    std::cout << "No extended version information found!\n";
+  }
```

Comment:
Would most of this function be more useful to place in the `support/Version.cpp.in` file so that other tools that use this library can call it without having to copy and paste from here?

CC @artemdinaburg

Reply:
What's the use case you're considering?

I wouldn't expect software built on top of SLEIGH to need to recreate this string in their own `--version` output.

Reply:
I think @artemdinaburg was wanting to integrate this information into a tool that would be able to print out not only the current tool's version but also the versions of the libraries/dependencies it used.

Okay, yeah, unless Artem says differently, we can just keep this here.